server/prisma/migrations/20240131062714_update_order_process/migration.sql

```diff
@@ -0,0 +1,212 @@
+ALTER TABLE "eventtickets" DROP CONSTRAINT "eventtickets_singleticket_fk_fkey";
+
+-- DropForeignKey
+ALTER TABLE "eventtickets" DROP CONSTRAINT "eventtickets_ticketrestrictionid_fk_fkey";
+
+-- DropForeignKey
+ALTER TABLE "eventtickets" DROP CONSTRAINT "tickets_eventinstanceid_fkey";
+
+-- DropForeignKey
+ALTER TABLE "orderitems" DROP CONSTRAINT "orderitems_orderid_fkey";
+
+-- DropForeignKey
+ALTER TABLE "orders" DROP CONSTRAINT "orders_contactid_fkey";
+
+-- DropForeignKey
+ALTER TABLE "orders" DROP CONSTRAINT "orders_discountid_fkey";
+
+-- DropForeignKey
+ALTER TABLE "orders" DROP CONSTRAINT "orders_orderdate_fkey";
+
+-- DropForeignKey
+ALTER TABLE "seasontickets" DROP CONSTRAINT "seasontickets_eventid_fk_fkey";
+
+-- DropForeignKey
+ALTER TABLE "seasontickets" DROP CONSTRAINT "seasontickets_eventticketid_fk_fkey";
+
+-- DropForeignKey
+ALTER TABLE "seasontickets" DROP CONSTRAINT "seasontickets_orderitemid_fk_fkey";
+
+-- DropForeignKey
+ALTER TABLE "seasontickets" DROP CONSTRAINT "seasontickets_seasontickettypeid_fk_fkey";
+
+-- DropForeignKey
+ALTER TABLE "singletickets" DROP CONSTRAINT "singletickets_orderitemid_fk_fkey";
+
+-- AlterTable
+ALTER TABLE "donations" DROP COLUMN "contactid_fk",
+DROP COLUMN "donationdate",
+DROP COLUMN "donorname",
+DROP COLUMN "isanonymous",
+DROP COLUMN "payment_intent",
+DROP COLUMN "refund_intent",
+ADD COLUMN     "anonymous" BOOLEAN NOT NULL DEFAULT false,
+ADD COLUMN     "orderid_fk" INTEGER NOT NULL,
+ALTER COLUMN "amount" SET NOT NULL,
+ALTER COLUMN "frequency" SET NOT NULL;
+
+-- AlterTable
+ALTER TABLE "eventinstances" DROP COLUMN "defaulttickettype";
+
+-- AlterTable
+ALTER TABLE "events" ALTER COLUMN "eventdescription" SET NOT NULL,
+ALTER COLUMN "eventdescription" SET DEFAULT '',
+ALTER COLUMN "active" SET NOT NULL,
+ALTER COLUMN "active" SET DEFAULT true,
+ALTER COLUMN "seasonticketeligible" SET NOT NULL,
+ALTER COLUMN "seasonticketeligible" SET DEFAULT true;
+
+-- AlterTable
+ALTER TABLE "orders" DROP COLUMN "orderdate",
+DROP COLUMN "ordertime",
+DROP COLUMN "ordertotal",
+DROP COLUMN "refund_intent",
+ADD COLUMN     "orderdateandtime" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP;
+
+-- AlterTable
+ALTER TABLE "seasons" ADD COLUMN     "deletedat" TIMESTAMP;
+
+-- AlterTable
+ALTER TABLE "ticketrestrictions" ADD COLUMN     "deletedat" TIMESTAMP(3);
+
+-- DropTable
+DROP TABLE "eventtickets";
+
+-- DropTable
+DROP TABLE "orderitems";
+
+-- DropTable
+DROP TABLE "seasontickets";
+
+-- DropTable
+DROP TABLE "seasontickettype";
+
+-- DropTable
+DROP TABLE "singletickets";
+
+-- CreateTable
+CREATE TABLE "order_ticketitems" (
+    "id" SERIAL NOT NULL,
+    "orderid_fk" INTEGER NOT NULL,
+    "price" MONEY NOT NULL,
+
+    CONSTRAINT "order_ticketitems_pkey" PRIMARY KEY ("id")
+);
+
+-- CreateTable
+CREATE TABLE "ticketitems" (
+    "id" SERIAL NOT NULL,
+    "order_ticketitemid_fk" INTEGER NOT NULL,
+    "ticketrestrictionid_fk" INTEGER NOT NULL,
+    "donated" BOOLEAN NOT NULL DEFAULT false,
```

Comment:
Do you know what the `donated` column means?

Reply:
There is a workflow were if a customer knows that they won't be able to make it to the show, instead of having the ticket refunded they can 'donate' it so that we don't count it against the number of remaining tickets. Functionally that means you can 'oversell' a show but that the physical number of people who show up wont ever be more then the set total of available tickets